src/bootman/kernel.c

```diff
@@ -684,6 +725,33 @@ bool boot_manager_install_kernel_internal(const BootManager *manager, const Kern
         return true;
 }
 
+bool _remove_extra_initrds(const char *initrd_base) {
+        autofree(char) *initrd_glob = string_printf("%s.*", initrd_base);
+        glob_t files;
+
+        int res = glob(initrd_glob, 0, NULL, &files);
+        if (res == GLOB_NOMATCH) {
+                globfree(&files);
+                return true;
+        }
+        if (res != 0) {
+                globfree(&files);
+                return false;
+        }
+
+        for(size_t i = 0; i < files.gl_pathc; i++) {
+                LOG_DEBUG("removing extra initrd: %s", files.gl_pathv[i]);
+
+                if (unlink(files.gl_pathv[i]) < 0) {
```

Comment:
This doesn't make sense to me and could introduce a FS accountancy error if we fail to remove a single file, yet remove others, due to breaking the loop. Suggestion: start with a `bool ret = true;` and if the unlink fails, set to `false`, so we can try to process *all* initrds.

Reply:
Good point. I changed the logic to what you suggested!